assets/js/base/hooks/cart/use-store-cart-item-quantity.js

```diff
@@ -62,9 +77,25 @@ export const useStoreCartItemQuantity = ( cartItem ) => {
 			changeCartItemQuantity( cartItemKey, debouncedQuantity );
 		}
 	}, [ debouncedQuantity, cartItemKey ] );
+	useEffect( () => {
+		if ( isPendingQuantity ) {
+			dispatchActions.incrementCalculating();
+		} else {
+			dispatchActions.decrementCalculating();
+		}
+	}, [ isPendingQuantity ] );
+
+	useEffect( () => {
+		if ( isPendingDelete ) {
+			dispatchActions.incrementCalculating();
+		}
+		return () => {
+			dispatchActions.decrementCalculating();
```

Comment:
running this directly in the return because the hook might unmount before `isPendingQuantity` change, so on unmount we're sure the element has been deleted.

Reply:
Yup this makes sense, the only problem is if the component _doesn't_ unmount because there was an error with the server removing. So you probably will need to account for whether there was a server error and make sure to decrement the count if `! isPendingDelete` and `whateverIsUsedToIndicateError` in the main body of the callback too.

Also `decrementCalculating` should only happen in the returned callback if `! isPendingDelete` otherwise if the component is unmounted for some other reason, the count could decrement unnecessarily.